torch/csrc/jit/codegen/cuda/interface.cpp

```diff
@@ -297,6 +297,117 @@ RegisterOperators reg_guard({
         aliasAnalysisFromSchema()),
 });
 
+
+bool checkView(
+    const at::Tensor& tensor,
+    const c10::TensorTypePtr& guard_tensor_type,
+    std::vector<int64_t> view_sizes,
+    c10::List<int64_t> constraint) {
+  // guard broadcast semantics, contiguity & stride order;
+  TORCH_INTERNAL_ASSERT(
+      guard_tensor_type && guard_tensor_type->dim().has_value());
+
+  // if num_dimension check fails or scalar type check fails
+  if (*guard_tensor_type->dim() != static_cast<size_t>(tensor.ndimension()) ||
+      *guard_tensor_type->dim() != constraint.size() ||
+      (guard_tensor_type->scalarType().has_value() &&
+       (guard_tensor_type->scalarType().value() != tensor.scalar_type()))) {
```

Comment:
Do we need to check for scalar_type for view?

Reply:
I think CudaFusionGuard should have done that already.

if that's the case, tensor input to the view guard can be changed to `List[int]`, which I think would make your life easier when we want to get to an intermediate tensor with a mutated size.